### ISSUE TITLE
Reveal existing buffer when opening file

### ISSUE THREAD

**Issue by tmandry** [zed-industries/zed#12572]
### Check for existing issues

- [X] Completed

### Describe the feature

Add an option to prefer revealing an existing buffer for a file when one is open instead of opening a new one. This would happen when navigating to a definition, clicking a file/line pair in the terminal, or opening a file from the file picker.

I like to have one copy of each file open and in a particular column. When navigating to an error or definition, if the file containing the definition is open in a buffer in another split, I would like for that buffer to be revealed and for the current file to stay visible in the current split.

This corresponds to the VSCode option `workbench.editor.revealIfOpen`.

**Comment by thesleort** [comment]
I very much would like to second this, as Zed becomes really cluttered if you have many splits and use <kbd>Ctrl</kbd>+<kbd>P</kbd> to navigate between files; and if the file is not already open, then it should open it in the currently active pane.

**Comment by Srekel** [comment]
Yes please! Here is a video of a case where I would like to click on a link to a file and not be taken to the already-open version of that file, especially when it's focused. 

In this case I also would assume that this bottom panel is not a "for editor buffers" type of panels and that new tabs should never open there. I'm not sure if there is something I can do as a user to set this behavior.


https://github.com/user-attachments/assets/f8a5aab7-0cd1-40ea-8159-662c8683a395

**Comment by WoodyWoodsta** [comment]
This is one of the biggest productivity slowdowns for me at the moment. If I have even 1 split, managing both buffers and which file is open in which becomes almost impossible to manage to the point of having to close all tabs and start again.

This should be something fairly simple to sort!
